sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -513,6 +513,14 @@ object SQLConf {
       .booleanConf
       .createWithDefault(false)
 
+  val MAX_SINGLE_PARTITION_BYTES = buildConf("spark.sql.maxSinglePartitionBytes")
+    .internal()
+    .doc("The maximum number of bytes allowed for a single partition. Otherwise, The planner " +
+      "will introduce shuffle to improve parallelism.")
+    .version("3.4.0")
+    .bytesConf(ByteUnit.BYTE)
+    .createWithDefaultString("128MB")
```

Comment:
For the safety, can we use more higher value, @wangyum ?

Reply:
I'd recommend to use `Long.MaxValue` like `spark.sql.defaultSizeInBytes` in Apache Spark 3.4.0. We can reduce it in 3.5.0 and later.

Reply:
ok